src/python/pants/option/global_options.py

```diff
@@ -200,12 +200,12 @@ def from_options(cls, options: Options) -> ExecutionOptions:
 
         # Determine the remote servers.
         # NB: Tonic expects the schemes `http` and `https`, even though they are gRPC requests.
-        # We have users set `grpc` and `grpcs` in the options system for clarity, but then
+        # We validate that users set `grpc` and `grpcs` in the options system for clarity, but then
         # normalize to `http`/`https`.
         remote_address_scheme = "https://" if bootstrap_options.remote_ca_certs_path else "http://"
         if bootstrap_options.remote_execution_address:
             remote_execution_address = bootstrap_options.remote_execution_address.replace(
-                "grpc", "http"
+                "grpc", "http", 1
```

Comment:
This assumes that the validation ran, which is fine. Although why not guarantee that only prefix changes by using `re.sub("^grpc", "http", bootstrap_options.remote_execution_address)`?